File: app/src/main/java/com/titolucas/mindlink/home/viewmodel/HomeViewModelFactory.java
package com.titolucas.mindlink.home.viewmodel;

import androidx.annotation.NonNull;
import androidx.lifecycle.ViewModel;
import androidx.lifecycle.ViewModelProvider;

import com.titolucas.mindlink.home.repository.HomeRepository;

public class HomeViewModelFactory implements ViewModelProvider.Factory {
    private final HomeRepository repository;

    public HomeViewModelFactory(HomeRepository repository) {
        this.repository = repository;
    }

    @NonNull
    @Override
    public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {
        if (modelClass.isAssignableFrom(HomeViewModel.class)) {
            return (T) new HomeViewModel(repository);
        }
        throw new IllegalArgumentException("Unknown ViewModel class");
    }
}

File: app/src/main/java/com/titolucas/mindlink/home/viewmodel/HomeViewModelFactory.kt
package com.titolucas.mindlink.home.viewmodel

import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider
import com.titolucas.mindlink.home.repository.HomeRepository

class HomeViewModelFactory(private val repository: HomeRepository) :
    ViewModelProvider.Factory {
    override fun <T : ViewModel> create(modelClass: Class<T>): T {
        if (modelClass.isAssignableFrom(HomeViewModel::class.java)) {
            return HomeViewModel(repository) as T
        }
        throw IllegalArgumentException("Unknown ViewModel class")
    }
}
